go/common/workerpool/workerpool.go

```diff
@@ -119,6 +160,15 @@ func (p *Pool) worker() {
 	defer p.workerGroup.Done()
 
 	for {
+		// Wait for the backoff period if configured.
+		if p.backoff != nil {
+			select {
+			case <-p.stopCh:
+				return
+			case <-time.After(p.backoff.Timeout()):
```

Comment:
Not sure if this works as intended. For example, if we have unlimited number of workers, it could happen that even if all jobs fail, we never wait before execution. But that is probably not relevant in the long run.

Reply:
Yeah, if a worker already waited for the `backoff.Timeout()` duration before it accepted a job, it won't delay further.

On a related note, I actually just noticed that the storage-sync worker pool is actually shared between all runtimes, so technically it is "mixing jobs A and B" and could potentially end up in a state that for one runtime all jobs are failing, while not for the other one. So I will probably move the backoff into the storage worker as you suggested before, and remove the backoff from the workerpool.